dracut/modules.d/99kiwi-dump-reboot/kiwi-dump-reboot-system.sh

```diff
@@ -33,17 +33,36 @@ function boot_installed_system {
     fi
 }
 
+function mount_sysroot {
+    local image_target
+    local root_uuid
+
+
+    [ -e /config.bootoptions ] || die "Missing config.bootoptions file"
+
+    root_uuid=$(
+        while read -r -d ' ' opt; do echo "${opt}";done < /config.bootoptions |\
+        grep root= | cut -f2- -d=
+    )
+
+    [ -z "${root_uuid}" ] && \
+        die "'root=' argument not found in config.bootoptions"
+
+
+    image_target=$(get_selected_disk)
+    kpartx -s -a "${image_target}"
```

Comment:
This part of the code looks mixed up and I got confused ;)

1. The reading of config.bootoptions is explicit to PXE deployments and done in code only if if getargbool 0 rd.kiwi.install.pxe; is true. This code imho doesn't belong here

2. The kpartx mapping of the devices happened as an exceptional case if getargbool 0 rd.kiwi.ramdisk; is true. Only in this case we need to directly map the partitions of the ramdisk block device after deployment. udev does not do this on ramdisk block devices. I also don't see this code to belong here

Reply:
> This part of the code looks mixed up and I got confused ;)
> 
>     1. The reading of config.bootoptions is explicit to PXE deployments and done in code only if if getargbool 0 rd.kiwi.install.pxe; is true. This code imho doesn't belong here

Probably this can be done differently, note this code is borrowed from the ramdisk generator and it is only used to get the `root=` argument from the `config.bootoptions` to generate the `sysroot.mount` unit required for ramdisk deployments. Probably we could load the `root_uuid` variable in previous stages and just test `root_uuid` exists here. I have no strong preference.

>     2. The kpartx mapping of the devices happened as an exceptional case if getargbool 0 rd.kiwi.ramdisk; is true. Only in this case we need to directly map the partitions of the ramdisk block device after deployment. udev does not do this on ramdisk block devices. I also don't see this code to belong here

I think the method name confused you, this method is ramdisk specific and it mounts the ramdisk, it is not being used in any other scenario. As you say we need kpartx here to get the ramdisk device mapped as a regular disk so we can use the syntax from `config.bootoptions` and refer to the root device by it's UUID.